Address object

wseifstad opened the pull request:
Created address object instead of pandas df

wseifstad committed:
Update README.md

wseifstad committed:
Created pandas dataframe for trash days, updated readme, created script for plotting address by trash days

wseifstad committed:
Merge branch 'master' of https://github.com/wseife/better-walk

wseifstad committed:
corrected map_plot html writing

wseifstad committed:
brought driver initiation out of the loop for faster performance

wseifstad committed:
implemented address objects, but still need database to push objects to

wseifstad committed:
deleted key

wyattshapiro commented on README.md:
```diff
@@ -1,16 +1,18 @@
 Clone the repo onto your machine with the following command:
 
-$ git checkout https://github.com/CloudburstCode/passiv.git
+$ git clone https://github.com/wseife/better-walk.git
```
lol whoops just copied that one over, good catch

wyattshapiro commented on src/address_with_days.csv:
```diff
@@ -0,0 +1,4759 @@
+address,trash_day
```
aweomse!

wyattshapiro commented on src/create_objects.py:
```diff
@@ -0,0 +1,62 @@
+    driver.close()
+    
+    #print test to show objects in list:
+    for i in object_list:
```
delete all these print lines since they are unnecessary

wyattshapiro commented on src/model.py:
```diff
@@ -0,0 +1,23 @@
+    address, latitude, longitude and trash_days
+    """
+
+    def __init__(self,address,lat,lng,trash_days):
```
perfect!

wyattshapiro commented on src/model.py:
```diff
@@ -0,0 +1,23 @@
+    def get_address(self):
+        return self.address
+
+    def get_lat(self):
```
delete

wyattshapiro commented on src/model.py:
```diff
@@ -0,0 +1,23 @@
+    def get_lat(self):
+        return self.lat
+    
+    def get_lng(self):
```
delete

wyattshapiro commented on src/model.py:
```diff
@@ -0,0 +1,23 @@
+    def get_lng(self):
+        return self.lng
+        
+    def get_trash_days(self):
```
delete

wseifstad replied:
Once we have a list of objects, don't we then need methods to call the object's address, lat, lng and days? I think I'm failing to see the benefit of using objects in this case as opposed to a table. In order to select a specific set of addresses (for example, that fall on a certain trash day), don't we need to call each object's trash_days method?

wyattshapiro commented on src/scrape.py:
```diff
@@ -1,35 +1,58 @@
         trash_days_dict = {}
         test_address = "160 15th Street, Brooklyn, NY, USA"
-        trash_days = scrape_util.getTrashDays(test_address, screenshot=True)
+        trash_days = scrape_util.getTrashDays(test_address, driver, screenshot=False)
```
word!

wyattshapiro commented on src/scrape.py:
```diff
@@ -1,35 +1,58 @@
+            print("checking address %s of %s" % (str(index+1) , len(address_list)))
+            print("program has been running for %s seconds" % str(time.time() - start_time))
+            trash_days = scrape_util.getTrashDays(address, driver, screenshot=False)
             trash_days_dict[address] = trash_days
```
we should be creating objects here, not a dictionary

wseifstad replied:
I'm going to delete scrape.py -- it's fully implemented in the create_objects.py script.

wyattshapiro commented on src/constants.py:
```diff
@@ -1 +1 @@
-BASE_DIR = "/Users/wyattshapiro/projects/dog_walk/"
+BASE_DIR = "/Users/williamseife/Documents/GitHub/better-walk/"
```
change this to the following:
`import os
BASE_DIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))`

This finds the real path of the file (constants.py), gets the directory (src/) and then gets that directory (better-walk/)

wseifstad replied:
done.

wyattshapiro commented on src/scrape.py:
```diff
@@ -1,35 +1,58 @@
-            trash_days_list.append(trash_days_dict)
+        
+        ## clear addresses with NoneType trash_days
+        clean_dict = {k: v for k, v in trash_days_dict.iteritems() if v is not None} 
```
why would we have a NoneType trash day? If we do, this should be functionalized and then called inside the getTrashDays

wseifstad replied:
This was the case for trash days that failed to scrape (i.e. if the address doesn't have any associated trash day, or if the browser fails to load the page). But, again, scrape.py should be ignored.

wyattshapiro commented on src/scrape.py:
```diff
@@ -1,35 +1,58 @@
+        ## clear addresses with NoneType trash_days
+        clean_dict = {k: v for k, v in trash_days_dict.iteritems() if v is not None} 
+
+        trash_df = pd.DataFrame.from_dict(clean_dict,orient="index") \
```
this should be a list of objects rather than dicts

wseifstad replied:
same as above

wyattshapiro commented on src/util/plot_util.py:
```diff
@@ -0,0 +1,41 @@
+        }  
+
+def geocode(address):
+    try:
```
word! try and except are great

wyattshapiro commented on src/create_objects.py:
```diff
@@ -0,0 +1,62 @@
+import pandas as pd
```
i am confused by this file, why are we plotting? 

we should be scraping in one function, then once we have scraped we can plot!

wseifstad replied:
bit of a misnomer with plot_util -- I created it for the map_plot, but it also contains the code for geocoding the addresses, hence my calling it for creating the Address objects. We can separate the geocoding into a different file, or rename this script if it's confusing.

wseifstad replied:
Here's what I'm going to do:
-rename "create_objects" this script "scrape.py"
functionality will be to scrape and geocode each address (throwing out addresses that fail to scrape or fail to geocode)
-move the geocode function into scrape_util to keep plotting separate.

wyattshapiro reviewed (changes requested):
Overall, the object looks great with the init function. Next step: in the scrape.py iterate though each address, scrape trashdays from getTrashDays, create an object for every unique address/day combo, store those list of objects in a master list of objects using .extend(). Finally, we can plot these address objects on a map.

wseifstad committed:
changed address_object.py to scrape.py and moved geocode to plot_util. Tidied formatting, changed BASE_DIR.

wseifstad committed:
added write to csv and changed API key storage

wseifstad committed:
Update README.md

wseifstad committed:
added standalone geocode script, for addresses that already have trash days. Added helper save to csv function in util. Created jupyter notebook for analysis of 11215 trash days.

wseifstad committed:
Merge branch 'address_object' of https://github.com/wseife/better-walk into address_object

wseifstad committed:
Add gif for readme

wseifstad committed:
map scatter plot (v2) for readme

wseifstad committed:
delete map scatter plot (v1)

wseifstad committed:
Updated Readme

Included gif showing trash day plots, and updated to reflect current functionality.
